ath79: add support for D-Link DIR-842 C3

pmelange opened the pull request:
Hardware spec of DIR-842 C3:
SoC: QCA9563
DRAM: 128MB DDR2
Flash: 16MB SPI-NOR
Switch: QCA8337N
WiFi 5.8GHz: QCA9888
WiFi 2.4Ghz: QCA9563
USB: circuit onboard, but components are not soldered

Flash instructions:

1. Upgrade the factory.bin through the factory web interface or
   the u-boot failsafe interface.
   The firmware will boot up correctly for the first time.
   Do not power off the device after OpenWrt has booted.
   Otherwise the u-boot will enter failsafe mode as the checksum
   of the firmware has been changed.
2. Upgrade the sysupgrade.bin in OpenWrt.
   After upgrading completes the u-boot won't complain about the
   firmware checksum and it's OK to use now.
3. If you powered off the device before upgrading the sysupgrade.bin,
   just upgrade the factory.bin through the u-boot failsafe interface
   and then goto step 2.

Signed-off-by: pmelange <isprotejesvalkata@gmail.com>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

blocktrron commented:
The Signed-Off-By line requires to contain your real name. Otherwise, this looks good to me.

adschm reviewed (approved):
LGTM

pmelange commented:
> The Signed-Off-By line requires to contain your real name. Same for your author name.

I've never noticed that before.  I guess it's some git command to set the author name.  I'll have to check it out.  

Nobody ever mentioned this to me before, and one or two of my commits have made it into LuCI, openwrt-routing, and maybe somewhere else.

By the way, I've installed the sysupgrade on the router and at first glance everything seems to be working fine, which isn't a big surprise.

adrianschmutzler commented:
> By the way, I've installed the sysupgrade on the router and at first glance everything seems to be working fine, which isn't a big surprise.

That's good, because besides the real name Signed-off it's also a prerequisite that your support patch is tested on device ;-)

pmelange commented:
Well, I just tried 

```
git config user.name "FIRST LAST"
git commit --amend
```
I then changed the signed-off line in the comment, but when I do a ```git log``` it still has ```Author: pmelange <isprotejesvalkata@gmail.com>```.  Any ideas?

pmelange committed:
ath79: add support for D-Link DIR-842 C3

Hardware spec of DIR-842 C3:
SoC: QCA9563
DRAM: 128MB DDR2
Flash: 16MB SPI-NOR
Switch: QCA8337N
WiFi 5.8GHz: QCA9888
WiFi 2.4Ghz: QCA9563
USB: circuit onboard, but components are not soldered

Flash instructions:

1. Upgrade the factory.bin through the factory web interface or
   the u-boot failsafe interface.
   The firmware will boot up correctly for the first time.
   Do not power off the device after OpenWrt has booted.
   Otherwise the u-boot will enter failsafe mode as the checksum
   of the firmware has been changed.
2. Upgrade the sysupgrade.bin in OpenWrt.
   After upgrading completes the u-boot won't complain about the
   firmware checksum and it's OK to use now.
3. If you powered off the device before upgrading the sysupgrade.bin,
   just upgrade the factory.bin through the u-boot failsafe interface
   and then goto step 2.

Signed-off-by: Perry Melange <isprotejesvalkata@gmail.com>

pmelange commented:
Figured it out...
```
git commit --reset-author  --amend
```

adrianschmutzler commented:
You can also specify the author as string with
`git commit --amend --author="something"`

blocktrron commented:
Thanks, merged.

pmelange commented:
There is a problem with flashing openwrt from the OEM firmware.  It would be really great if the instructions posted here: https://forum.openwrt.org/t/d-link-dir-842-c3-installation/43094/7 were added to https://openwrt.org/toh/d-link/d-link_dir-842
